flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -395,12 +405,20 @@ public void onContainersAllocated(List<Container> containers) {
 						nodeManagerClient.startContainer(container, taskExecutorLaunchContext);
 					} catch (Throwable t) {
 						log.error("Could not start TaskManager in container {}.", container.getId(), t);
-
 						// release the failed container
 						workerNodeMap.remove(resourceId);
 						resourceManagerClient.releaseAssignedContainer(container.getId());
-						// and ask for a new one
-						requestYarnContainerIfRequired();
+						log.error("Could not start TaskManager in container {}.", container.getId(), t);
+						recordFailure();
+						if (shouldRejectRequests()) {
+							rejectAllPendingSlotRequests(new MaximumFailedTaskManagerExceedingException(
+								String.format("Maximum number of failed container %d in interval %s"
+										+ "is detected in Resource Manager", maximumFailureTaskExecutorPerInternal,
+									failureInterval.toString()), t));
```

Comment:
This branch should go into the `recordFailure` method.

Reply:
done